### PR TITLE
[Add] Support loading data from ceph for mot_challenge_dataset.

### PR DESCRIPTION
**Pull request by pixeli99**

### CODE REVIEW

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 1]
```diff
@@ -374,12 +373,34 @@ def evaluate(self,
                 else:
                     gt_file = osp.join(self.img_prefix, f'{name}/gt/gt.txt')
                 res_file = osp.join(resfiles['track'], f'{name}.txt')
-                gt = mm.io.loadtxt(gt_file)
+                # copy gt file from ceph to local temporary directory
+                # no 'self.img_prefix'
+                gt_dir_path = osp.join(local_dir.name, name, 'gt')
+                os.makedirs(gt_dir_path)
+                gt_file_tmp = local_dir.name + gt_file.replace(
```
Comment (lpxgis) [inline mmtrack/datasets/mot_challenge_dataset.py:380]: Using `+` may be necessary, the `local_dir.name` is `/tmp/*******` and the `gt_file.replace` is `/mot_name/gt/gt.txt`. Using `osp.join` may be a dangerous operation, and `os.sep` should also be unnecessary. I wonder if you have some suggestions?

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 2]
```diff
@@ -22,7 +22,6 @@
 @DATASETS.register_module()
 class MOTChallengeDataset(CocoVideoDataset):
     """Dataset for MOTChallenge.
-
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:25]: usually we have a blank line in here. Please revert the change

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 3]
```diff
@@ -362,8 +358,11 @@ def evaluate(self,
                 results, resfile_path, metrics)
             print_log('Evaluate CLEAR MOT results.', logger=logger)
             distth = 1 - track_iou_thr
-
             accs = []
+            # support loading data from ceph
+            client = self.file_client
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:363]: delete this line
directly using self.file_client rather than cilent

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 4]
```diff
@@ -403,16 +424,16 @@ def evaluate(self,
                 f.write('name\n')
                 for name in names:
                     f.write(name + '\n')
-                f.close
+                f.close()
 
             eval_config = trackeval.Evaluator.get_default_eval_config()
 
+            gt_folder = local_dir.name
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:431]: delete this line

File: mmtrack/datasets/mot_challenge_dataset.py (hunk 1 above)
Comment (GT9505) [reply mmtrack/datasets/mot_challenge_dataset.py:380]: I still don't get why we can't use osp.join?

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 5]
```diff
@@ -89,7 +89,6 @@ def _parse_ann_info(self, img_info, ann_info):
         Args:
             ann_info (list[dict]): Annotation info of an image.
             with_mask (bool): Whether to parse mask annotations.
-
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:92]: Please follow [this](https://github.com/open-mmlab/mmtracking/blob/master/mmtrack/models/sot/stark.py#L81-L90) to change the format of docstring.
This PR miss a log of blank lines in docstring.

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 6]
```diff
@@ -374,12 +373,35 @@ def evaluate(self,
                 else:
                     gt_file = osp.join(self.img_prefix, f'{name}/gt/gt.txt')
                 res_file = osp.join(resfiles['track'], f'{name}.txt')
-                gt = mm.io.loadtxt(gt_file)
+                # copy gt file from ceph to local temporary directory
+                # no 'self.img_prefix'
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:377]: `no 'self.img_prefix'` can be deleted

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 7]
```diff
@@ -374,12 +373,35 @@ def evaluate(self,
...
+                    gt_file.replace(gt_file.split(name)[0], ''))
+
+                f = open(gt_file_tmp, 'wb')
+                gt_content = self.file_client.get(gt_file)
+                if hasattr(gt_content, 'tobytes'):
+                    gt_content = gt_content.tobytes()
+                f.write(gt_content)
+                f.close()
+                # copy sequence file from ceph to local temporary directory
+                seqinfo_path = osp.join(local_dir.name, name, 'seqinfo.ini')
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:391]: The `seqinfo.ini` may not exist.
Therefore, we need to copy `seqinfo.ini` only when it exist.

Comment (lpxgis) [reply mmtrack/datasets/mot_challenge_dataset.py:394]: But this file maybe necessary, because it will be used again on line **442**.
`dataset = [trackeval.datasets.MotChallenge2DBox(dataset_config)]`
If we don't throw errors here, we will still throw errors later.

**Review by GT9505 — COMMENTED** [review]
We can merge this pr after making sure that we support loading data from both ceph and local disk.

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 8]
```diff
@@ -374,12 +377,34 @@ def evaluate(self,
                 else:
                     gt_file = osp.join(self.img_prefix, f'{name}/gt/gt.txt')
                 res_file = osp.join(resfiles['track'], f'{name}.txt')
-                gt = mm.io.loadtxt(gt_file)
+                # copy gt file from ceph to local temporary directory
+                gt_dir_path = osp.join(local_dir.name, name, 'gt')
+                os.makedirs(gt_dir_path)
+                gt_file_tmp = osp.join(
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:383]: gt_file_tmp -> copied_gt_file

File: mmtrack/datasets/mot_challenge_dataset.py [hunk 9]
```diff
@@ -374,12 +377,34 @@ def evaluate(self,
...
+                    gt_file.replace(gt_file.split(name)[0], ''))
+
+                f = open(gt_file_tmp, 'wb')
+                gt_content = self.file_client.get(gt_file)
+                if hasattr(gt_content, 'tobytes'):
+                    gt_content = gt_content.tobytes()
+                f.write(gt_content)
+                f.close()
+                # copy sequence file from ceph to local temporary directory
+                seqinfo_path = osp.join(local_dir.name, name, 'seqinfo.ini')
```
Comment (GT9505) [inline mmtrack/datasets/mot_challenge_dataset.py:394]: seqinfo_path -> copied_seqinfo_path

**Comment by pixeli99** [comment]
> We can merge this pr after making sure that we support loading data from both ceph and local disk.
The test is successful on MOT16 and MOT17, using the faster rcnn model.

**Review by GT9505 — APPROVED** [review]
